Docs: Relabel and move explanation and how-tos around OAuth (Diátaxis) (#9834)
readthedocs/readthedocs.org: docs/user/guides/connecting-git-account.rst

How to connect your Read the Docs account to your Git repository
================================================================

If you are going to import repositories from GitHub, Bitbucket, or GitLab,
you should connect your Read the Docs account to your repository host first.

.. note::

   If you signed up or logged in to Read the Docs with your GitHub, Bitbucket, or GitLab credentials,
   you're all done. Your account is connected ✅️.

> To connect a social account, go to your :guilabel:`<Username dropdown>` > :guilabel:`Settings` > :guilabel:`Connected Services`.

Reviewer: This feels underwhelming. We should have some other kind of content here, or a video? I think overall we should probably be sharing a couple benefits or something though, along with the basic process.

Author: I'll be doing a few updates on this one. I'm writing this comment to reflect on the goals of having a how-to here since it's border-line.

Goals of these changes:

* Showcase that it's _easy_ to add the integration
* Add screenshots that will make it easier to skim the how-to (because the texts are a bit trivial)
* Declare when this how-to is relevant

How to connect your Read the Docs account to your Git provider
==============================================================

> In this how-to,

Reviewer: I feel like we aren't generally using `how-to`

```suggestion
In this how to article,
```

> you are shown the steps to connect an account on GitHub, Bitbucket, or GitLab with your Read the Docs account.

Reviewer: ```suggestion
you are shown the steps to connect an account on |git_provider_and| with your Read the Docs account.
```

:D

Author: Exactly this! I've just merged the style guide changes to `main`, then to `diataxis/main` and then into this PR so I can use the substitutions here :)

or if you have signed up using a Git provider account and want to connect additional providers.

If you are going to import repositories from GitHub, Bitbucket, or GitLab,
you should connect your Read the Docs account to your Git provider first.

.. note::

   If you signed up or logged in to Read the Docs with your GitHub, Bitbucket, or GitLab credentials,
   you're all done. Your account is connected ✅️.
   You only need this how-to if you want to connect additional Git providers.

Adding a connection
-------------------

To connect your Read the Docs account with a Git provider,
go to the main login menu: :guilabel:`<Username dropdown>` > :guilabel:`Settings` > :guilabel:`Connected Services`.

From here, you'll be able to connect to your GitHub, Bitbucket or GitLab
account. This process will ask you to authorize an integration with Read the Docs.

.. figure:: /img/oauth_github_dialog.png
   :width: 300px
   :align: center
   :alt: Screenshot of example OAuth dialog on GitHub

   An example of how your OAuth dialog on GitHub may look.

After approving the request,
you will be taken back to Read the Docs.
You will now see the account appear in the list of connected services.

> .. figure:: /img/screenshot_connected_services.png

Reviewer: Love Donald Docs as a test user 😆
